hashintel/hash
# H-1781: Add new users to updates list

## 🌟 What is the purpose of this PR?




This PR adds the ability for adding HASH users in production to a mailchimp list, specifying their email, shortname, preferredName, and that they registered with HASH.

## 🔗 Related links





- H-1769

## 🔍 What does this change?




- adds after create/update user hooks that create or update a user in mailchimp via an email address

## Pre-Merge Checklist 🚀

### 🚢 Has this modified a publishable library?




This PR:


### 📜 Does this require a change to the docs?




The changes in this PR:


### 🕸️ Does this require a change to the Turbo Graph?




The changes in this PR:


## ⚠️ Known issues



- We want to support setting more than one email for a user, which means we need to come up with a way of deciding which emails to use to subscribe to mailchimp.

## 🐾 Next steps



## 🛡 What tests cover this?



Manual testing.

## ❓ How to test this?



1. Add the `MAILCHIMP_LIST_ID` and `MAILCHIMP_API_KEY` environment variables to your local developer environment. These can be obtained in the mailchimp UI, or by reading [the environment variables in vault](https://vault.blockprotocol.org/ui/vault/secrets/automation/show/pipelines/hash/prod)
1. Run HASH in production mode locally
2. Sign up with a real email (`alice@example.com` is detected as a fake email by mailchimp)
3. Verify [in the mailchimp UI](https://us15.admin.mailchimp.com/lists/members?id=61455) that the user is added with the correct fields

## 📹 Demo



Creating the following user:

<img width="1050" alt="image" src="https://github.com/hashintel/hash/assets/42802102/bc645be6-caaa-4fa9-b2b7-b5212d59e8c3">

Results in the following mailchimp user:

<img width="603" alt="image" src="https://github.com/hashintel/hash/assets/42802102/9de4935d-6273-4429-baef-898a28565d8e">
